packages/pglite/src/utils.ts

```diff
@@ -174,3 +174,55 @@ export async function formatQuery(
   )
   return ret.rows[0].query
 }
+
+/**
+ * Debounce a function to ensure that only one instance of the function is running at
+ * a time.
+ * - If the function is called while an instance is already running, the new
+ * call is scheduled to run after the current instance completes.
+ * - If there is already a scheduled call, it is replaced with the new call.
+ * @param fn - The function to debounce
+ * @returns A debounced version of the function
+ */
+export function debounceMutex<A extends any[], R>(
```

Comment:
the name has burned my brain haha
might be worth adding a test or two as a sanity check cause this function's behaviour is not trivially obvious